### PR TITLE
feat: add product crud operations

### PR DESCRIPTION
**Pull request by machi1990**
##### Checklist

- [ ] `npm test` passes
- [ ] `npm run build` works
- [ ] tests are included
- [ ] documentation is changed or added

### CODE REVIEW

**Comment by machi1990** [comment]
- I initially wanted to have one component / page to handle the view or creation of the product but had issues with react hooks as I was using some hooks conditionally. I went with the easier approach and decided to split them into two  View and Create Product Page. 
- It is possible to create a product without distribution centre, i.e `distributionCentreId: undefined` but when you retrieve it from, it has a `distributionCentre` assigned. Kinda strange but I think it is an issue on graphback as a result of this filter https://github.com/aerogear/OpenVolunteerPlatform/pull/62/files#diff-3cd51889b484d5854b87584c0b8e0b6fR29 where `id= undefined`. Have not dig further though. cc @wtrocki @craicoverflow 
- With this query https://github.com/aerogear/OpenVolunteerPlatform/pull/62/files#diff-7ab02896187348b375fbe6ee7eadace6R12, I would love to only fetch `id` and `name` to avoid over fetching of fields that I do not want. Is it possible?

File: platform/client-admin/src/forms/product.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+const productSchema = new SimpleSchema({
+    label: {
+        type: String,
+        max: 50
+    },
+    description: {
+        type: String,
+        max: 500
+    },
+    distributionCentre: {
```
Comment (machi1990) [inline platform/client-admin/src/forms/product.ts:13]: Working on making this selectable.

**Comment by wtrocki** [comment]
> I initially wanted to have one component / page to handle the view or creation of the product but had issues with react hooks as I was using some hooks conditionally

We have two versions of hooks. Normal and Lazy. Lazy hooks can be done conditionally.

**Comment by wtrocki** [comment]
> With this query #62 (files)

You can create your own queries and as long you not going to remove entire folder all will be fine. Fragments are optional so queries can be issued with anything you want (and graphql-code-generator will pick that up). That is why we have 2 level generation now

**Comment by wtrocki** [comment]
> It is possible to create a product without distribution centre,  

There is no magic in software :)
Please share your query. As for linked code distribution centre id needs to be hardcoded or supplied somewhere.

**Comment by machi1990** [comment]
> > I initially wanted to have one component / page to handle the view or creation of the product but had issues with react hooks as I was using some hooks conditionally
> 
> We have two versions of hooks. Normal and Lazy. Lazy hooks can be done conditionally.

Yes, I have seen them and tried to use them but it ended up being a lot of code in one page that I have decided to just split into two separate components each one handling separate concerns.

**Comment by machi1990** [comment]
Oops, clicked the wrong button and edited your comment lol

> > With this query #62 (files)
> 
> You can create your own queries and as long you not going to remove entire folder all will be fine. Fragments are optional so queries can be issued with anything you want (and graphql-code-generator will pick that up). That is why we have 2 level generation now

Yeap, I thought maybe there is an option I could pass into the generated query. Anyway, let me create my own query, could be fun.

**Comment by machi1990** [comment]
> > It is possible to create a product without distribution centre,
> 
> There is no way graphback will assign something on the backend.
> I will look for issue in the client.

Well, I am not sure if it is client related since I observed the behaviour with the playground too.

> Please share your query. As for linked code distribution centre id needs to be hardcoded or supplied somewhere.

To reproduce, checkout the PR locally. 

Start the server and create a product with the following mutation:
```
mutation createProductMutation {
  createProduct(input: {
    label: "Test", 
    description: "This is a test product that is missing a distribution centre"}) {
    id
  }
}
```

Retrieve the list of created products with:
```
query {
  findAllProducts {
    id,
    label,
    description,
    distributionCentre {
      id,
      name
    }
  }
}
```

I would expect the created newly created product to not have a `distributionCentre` but it has.
```json
"distributionCentre": {
     "id": "5ea305da65c3575d036ae57b",
     "name": "NY Plaza Charity Chain"
}
```

**Pull request merged**